### PR TITLE
Add support for Cortex-R7

### PR DESCRIPTION
**Pull request by julien-massot**
This is a preliminary pull request in order to add support for Renesas R-Car Gen3 based board.
Most of Renesas R-Car Gen3 based board contains a cortex-r7 processor which is able to run Zephyr.
Add definition for the Cortex R7 before submitting the SoC and board supports.

### CODE REVIEW

File: dts/bindings/cpu/arm,cortex-r7.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2020 IoT.bzh
```
Comment (carlocaione) [inline dts/bindings/cpu/arm,cortex-r7.yaml:1]: 2021?

Comment (julien-massot) [reply dts/bindings/cpu/arm,cortex-r7.yaml:1]: Indeed, will refresh the Copyright thanks.

**Review by carlocaione — APPROVED** [review]

**Review by mbolivar-nordic — APPROVED** [review]

**Comment by ioannisg** [comment]
Should this have a dependence on the CMSIS module update?

**Review by ioannisg — APPROVED** [review]

**Comment by julien-massot** [comment]
> Should this have a dependence on the CMSIS module update?
> 
> I mean, this. [zephyrproject-rtos/cmsis#8](https://github.com/zephyrproject-rtos/cmsis/pull/8)
> 
> (I don't think so but just asking, @julien-massot)

Hi, good point, my plan to to submit soon a pull request which have a dependency on the CMSIS module update.
So that's for the next one.

So this PR doesn't have any dependencies

**Pull request merged**